modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -85,7 +91,10 @@ CV__DNN_INLINE_NS_BEGIN
         DNN_TARGET_OPENCL_FP16,
         DNN_TARGET_MYRIAD,
         DNN_TARGET_VULKAN,
-        DNN_TARGET_FPGA  //!< FPGA device with CPU fallbacks using Inference Engine's Heterogeneous plugin.
+        DNN_TARGET_FPGA,  //!< FPGA device with CPU fallbacks using Inference Engine's Heterogeneous plugin.
+        DNN_TARGET_CUDA_FP16,
+        DNN_TARGET_CUDA_FP32,
+        DNN_TARGET_CUDA_FP64
```

Comment:
I was wondering about the use-case of the FP64 target. It helps with debugging for a developer but I think it isn't of much use to the end-user.

The DNN backbone code doesn't allow the user to submit FP64 blobs as input. Hence, there simply is no point in having FP64 target.

It can be easily added in the future. It doesn't take more than 30 minutes to add a new target (just have to instantiate a ton of templates in several files and add some specalizations for vector types and math functions).

Reply:
I wouldn't bother adding it.  FP64 isn't something anyone uses for DNNs.

Reply:
I think that without `DNN_TARGET_CUDA_FP64` we could rename `DNN_TARGET_CUDA_FP32` to just `DNN_TARGET_CUDA` (in the same way as we have for OpenCL: `DNN_TARGET_OPENCL` and `DNN_TARGET_OPENCL_FP16`).